var/spack/repos/builtin/packages/bufr/package.py

```diff
@@ -11,37 +13,108 @@ class Bufr(CMakePackage):
+        return (
+            f"https://github.com/NOAA-EMC/NCEPLIBS-bufr/archive/refs/tags/{pre}v{version}.tar.gz"
+        )
+
+    # Need to make the lines shorter at least on some systems
+    def patch(self):
+        with when("@:11.7.1"):
+            filter_file("_lenslmax 120", "_lenslmax 60", "CMakeLists.txt")
+
+    def cmake_args(self):
+        args = [
+            self.define_from_variant("ENABLE_PYTHON", "python"),
+            self.define_from_variant("BUILD_SHARED_LIBS", "shared"),
+            self.define("BUILD_TESTS", self.run_tests),
+        ]
+
+        return args
+
+    def flag_handler(self, name, flags):
+        """
+        On macOS if a library built with the ar utility contains objects
+        with Fortran module data but no executable functions,
+        the symbols corresponding to the module data may not be resolved
+        when an object referencing them is linked against the library.
+        You can work around this by compiling with option -fno-common.
+        """
+        fc = self.compiler.fc
+        if self.spec.satisfies("platform=darwin"):
+            if name == "fflags":
+                if "ifort" in fc or "gfortran" in fc:
+                    flags.append("-fno-common")
+
+        # Bufr inserts a path into source code which may be longer than 132
+        if name == "fflags" and "gfortran" in fc:
+            flags.append("-ffree-line-length-none")
+
+        # Inject flags into CMake build
+        return (None, None, flags)
 
     def _setup_bufr_environment(self, env, suffix):
         libname = "libbufr_{0}".format(suffix)
-        lib = find_libraries(libname, root=self.prefix, shared=False, recursive=True)
-        lib_envname = "BUFR_LIB{0}".format(suffix)
-        inc_envname = "BUFR_INC{0}".format(suffix)
-        include_dir = "include_{0}".format(suffix)
+        shared = True if "+shared" in self.spec else False
+        # Bufr has _DA (dynamic allocation) libs in versions <= 11.5.0
+        append = "" if self.spec.satisfies("@11.5.0:") else "_DA"
+        lib = find_libraries(libname + append, root=self.prefix, shared=shared, recursive=True)
+        lib_envname = "BUFR_LIB{0}".format(suffix) + append
+        inc_envname = "BUFR_INC{0}".format(suffix) + append
+        include_dir = "{0}_{1}".format(self.prefix.include.bufr, suffix)
 
         env.set(lib_envname, lib[0])
         env.set(inc_envname, include_dir)
 
-        # Bufr has _DA (dynamic allocation) libs in versions <= 11.5.0
-        if self.spec.satisfies("@:11.5.0"):
-            da_lib = find_libraries(
-                libname + "_DA", root=self.prefix, shared=False, recursive=True
-            )
-            env.set(lib_envname + "_DA", da_lib[0])
-            env.set(inc_envname + "_DA", include_dir)
+        if self.spec.satisfies("+python"):
+            pyver = self.spec["python"].version.up_to(2)
+            pydir = os.path.join(os.path.dirname(lib[0]), "python%s/site-packages" % pyver)
```

Comment:
You're already using the path join so why not include the last directory node?  You can also use an f-string.

```suggestion
            pydir = join_path(os.path.dirname(lib[0]), f"python{pyver}", "site-packages")
```